pkg/util/time.go

```diff
@@ -35,6 +36,19 @@ func FormatTimeModel(t model.Time) string {
 	return TimeFromMillis(int64(t)).String()
 }
 
+// ParseTimeParam parses the desired time param from a Prometheus http request into an int64, milliseconds since epoch.
+func ParseTimeParam(r *http.Request, paramName string, defaultValue int64) (int64, error) {
```

Comment:
like `ParseTime` below, this just lifts a prometheus util out of https://github.com/grafana/mimir/blob/41af1393d6e2b799c701448f7c54c62c15075916/vendor/github.com/prometheus/prometheus/web/api/v1/api.go
and updates it to use int64 unix milliseconds instead of time.Time